nautobot/extras/utils.py

```diff
@@ -80,6 +82,43 @@ def get_choices(self):
         return [(f"{ct.app_label}.{ct.model}", ct.pk) for ct in ContentType.objects.filter(self.get_query())]
 
 
+@deconstructible
+class ModelSubclassesQuery:
+    """
+    Helper class to get ContentType models that are subclasses of self.klass
+    """
+
+    def __init__(self, klass):
+        self.klass = klass
+
+    def list_subclasses(self):
+        """
+        Return a list of classes that inherits from self.klass
+        """
+        klass = import_object(self.klass)
+        return [_class for _class in apps.get_models() if issubclass(_class, klass)]
+
+    def __call__(self):
+        return self.get_query()
+
+    @property
+    def as_queryset(self):
+        return ContentType.objects.filter(self.get_query()).order_by("app_label", "model")
+
+    def get_query(self):
+        """
+        Given an extras feature, return a Q object for content type lookup
+        """
+        query = Q()
+        for model in self.list_subclasses():
+            query |= Q(app_label=model._meta.app_label, model=model.__name__.lower())
+
+        return query
```

Comment:
This appears to be three ways to do the same thing. There are some cases where `ModelSubclassesQuery("foo").as_queryset` or `.get_query` appear to be used interchangeably, and then there are a couple places where `ModelSubclassesQuery("foo")` is provided by itself, without any attribute lookups.

Can we please streamline this? My preference would be to just defer to `ModelSubclassesQuery("foo")` and fold the others into it.
```suggestion
    def __call__(self):
        """
        Given an extras feature, return a Q object for content type lookup
        """
        query = Q()
        for model in self.list_subclasses():
            query |= Q(app_label=model._meta.app_label, model=model.__name__.lower())

        return query

    @property
    def as_queryset(self):
        return ContentType.objects.filter(self()).order_by("app_label", "model")
```

Reply:
Removing the monkey-patching and rolling forward, yes.